sos/report/plugins/k8s.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2024 Canonical Ltd., Arif Ali <arif.ali@canonical.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+
+class K8s(Plugin, UbuntuPlugin):
+    """The K8s plugin collects the current status of the k8s snap on a
+    Ubuntu machine.
+
+    It will collect logs from journald related to the snap.k8s units as
+    well as run k8s commands to retrieve the configuration, status.
+    """
+
+    short_desc = 'Canonical Kubernetes'
+    plugin_name = "k8s"
+    profiles = ('container',)
+
+    packages = ('k8s',)
+
+    k8s_cmd = "k8s"
+
+    def setup(self):
+        self.add_journal(units="snap.k8s.*")
+
+        k8s_common = "/var/snap/k8s/common"
+
+        k8s_subcmds = [
+            'status',
+            'get',
+        ]
+
+        self.add_copy_spec([
+            "/etc/kubernetes/"
+            f"{k8s_common}/args",
+            f"{k8s_common}/var/lib/k8s-dqlite/info.yaml",
+            f"{k8s_common}/var/lib/k8s-dqlite/cluster.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/truststore/k8s.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/database/info.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/database/cluster.yaml",
+            f"{k8s_common}/var/lib/k8sd/state/daemon.yaml",
```

Comment:
The `yaml` files cant contain any credentials worth to `postproc`, right?

Reply:
I have gone through these files maually twice :), and there isn't anything to obfuscate, except for `/var/snap/k8s/common/var/lib/k8sd/state/truststore/k8s.yaml`, which I've already added in `postproc()`

I am now getting all of this verified by our k8s SMEs to ensure we got everything